src/main/java/org/simulator/sbml/SimpleConstraintListener.java

```diff
@@ -58,7 +65,22 @@ public void processViolation(ConstraintEvent evt) {
     // Math must be set, otherwise this event would not have been triggered.
     constraint = evt.getSource().getMath().toFormula();
     message = SBMLtools.toXML(evt.getSource().getMessage());
-    // TODO: Localize
-    logger.warn(MessageFormat.format("[VIOLATION]\t{0} at time {1,number}: {2}", constraint, evt.getTime(), message));
+
+    if (evt.getSource().getUserObject(CONSTRAINT_VIOLATION_LOG) == null) {
+      evt.getSource().putUserObject(CONSTRAINT_VIOLATION_LOG, Boolean.FALSE);
+    }
+
+    boolean constraintCondition = ((ASTNodeValue) evt.getSource().getMath().getUserObject(TEMP_VALUE)).compileBoolean(evt.getTime());
```

Comment:
When we reach this line of code, the constraint's condition was causing the listener to become active. Re-evaluation of the condition is unnecessary.